python_modules/dagster/dagster/core/asset_defs/assets.py

```diff
@@ -396,9 +399,29 @@ def get_resource_requirements(self) -> Iterator[ResourceRequirement]:
     def required_resource_keys(self) -> Set[str]:
         return {requirement.key for requirement in self.get_resource_requirements()}
 
+    def to_string(self):
```

Comment:
Any particular reason to call this `to_string` instead of `__str__`. Even though we use `to_string` in some places in our codebase, I think it's generally considered better practice to use `__str__`.

Reply:
potential user impact from users calling str(assets_def) is what I was thinking, but if that's not a concern I can change.